File: src/main/java/pl/training/workshop/m8/s11_stagedrollout/start/CheckoutRouter.java
package pl.training.workshop.m8.s11_stagedrollout.start;

/**
 * Start: wdrożenie nowego procesu płatności "na flagę". Stała w kodzie (zmiana = nowe wydanie),
 * lista testerów wpisana w if, brak podziału na etapy i brak wyłącznika awaryjnego.
 */
public final class CheckoutRouter {
    static final boolean NEW_CHECKOUT = false;

    public boolean useNewCheckout(String email) {
        if (NEW_CHECKOUT) {
            return true;
        }
        return email.equals("anna@kino.pl") || email.equals("jan@kino.pl");
    }
}

File: csharp/src/Training.Workshop/M8/S11StagedRollout/Start/CheckoutRouter.cs
namespace Training.Workshop.M8.S11StagedRollout.Start;

/// <summary>
/// Start: wdrożenie nowego procesu płatności "na flagę". Stała w kodzie (zmiana = nowe wydanie),
/// lista testerów wpisana w if, brak podziału na etapy i brak wyłącznika awaryjnego.
/// </summary>
public sealed class CheckoutRouter
{
    internal static readonly bool NewCheckout = false;

    public bool UseNewCheckout(string email)
    {
        if (NewCheckout)
        {
            return true;
        }
        return email.Equals("anna@kino.pl") || email.Equals("jan@kino.pl");
    }
}
